code/ATMOSPHERICS/components/unary/cap.dm

```diff
@@ -125,3 +125,61 @@
 		return // Coloring pipes.
 
 	return ..()
+
+/obj/machinery/atmospherics/unary/cap/bluespace
+	name = "bluespace pipe endcap"
+	desc = "A bluespace endcap for pipes."
+	icon = 'icons/obj/pipes.dmi'
+	icon_state = "bscap"
+	can_be_coloured = 0
+	var/global/list/obj/machinery/atmospherics/bspipe_list = list()
+	
+/obj/machinery/atmospherics/unary/cap/bluespace/blue
+	color=PIPE_COLOR_BLUE
+	
+/obj/machinery/atmospherics/unary/cap/bluespace/cyan
+	color=PIPE_COLOR_CYAN
+	
+/obj/machinery/atmospherics/unary/cap/bluespace/green
+	color=PIPE_COLOR_GREEN
+	
+/obj/machinery/atmospherics/unary/cap/bluespace/pink
+	color="#ff66cc"
+	
+/obj/machinery/atmospherics/unary/cap/bluespace/purple
+	color=PIPE_COLOR_PURPLE
+	
+/obj/machinery/atmospherics/unary/cap/bluespace/red
+	color=PIPE_COLOR_RED
+	
+/obj/machinery/atmospherics/unary/cap/bluespace/orange
+	color=PIPE_COLOR_ORANGE
+	
+/obj/machinery/atmospherics/unary/cap/bluespace/update_icon()
+	overlays = 0
+	alpha = invisibility ? 128 : 255
+	icon_state = "bscap"
+	
+/obj/machinery/atmospherics/unary/cap/bluespace/New()
+	..()
+	bspipe_list.Add(src)
+	
+/obj/machinery/atmospherics/unary/cap/bluespace/Destroy()
+	bspipe_list.Remove(src)
+	..()
+	
+/obj/machinery/atmospherics/unary/cap/bluespace/proc/merge_all()
+	var/datum/pipe_network/main_network
+	for(var/obj/machinery/atmospherics/unary/cap/bluespace/bscap in bspipe_list)
+		if(!main_network && bscap.network && (src.color == bscap.color))
+			main_network = bscap.network
+			continue
+		if(main_network && bscap.network && (src.color == bscap.color))
+			main_network.merge(bscap.network)
```

Comment:
main_network is always going to be null since you just initialized it on line 172, so your second if statement likely is never accessed.

Reply:
main_network is set in line 175.

Reply:
I know; but any time merge_all() is called 'main_network' will be cleared via line 172. Then once you iterate through all bscap's in bspipe_list, they will only hit the first if statement which will then move onto the next item in the list since you have a continue here. Since the for loop only iterates across bspipe_list once, you will never hit the second if statement.